src/main/java/io/cdap/e2e/utils/ElementHelper.java

```diff
@@ -0,0 +1,135 @@
+  }
+
+  public static void clickOnElement(WebElement element) {
+    scrollToElement(element);
+    WaitHelper.waitForElementToBeClickable(element);
+    logger.info("Click on the element: " + element);
+    element.click();
+  }
+
+  public static void clickOnElement(WebElement element, long timeoutInSeconds) {
+    scrollToElement(element);
+    WaitHelper.waitForElementToBeClickable(element, timeoutInSeconds);
+    logger.info("Click on the element: " + element);
+    element.click();
+  }
+
+  public static void clickOnElementUsingJsExecutor(WebElement element) {
+    logger.info("Click on the element using Javascript Executor: " + element);
+    js.executeScript("arguments[0].click();", element);
+  }
+
+  public static void sendKeys(WebElement element, String keys) {
+    scrollToElement(element);
+    WaitHelper.waitForElementToBeDisplayed(element);
+    logger.info("Send keys to element: " + element);
+    element.sendKeys(keys);
+  }
+
+  public static void hoverOverElement(WebElement element) {
+    scrollToElement(element);
+    WaitHelper.waitForElementToBeDisplayed(element);
+    logger.info("Hovering over the element: " + element);
+    actions.moveToElement(element).build().perform();
+  }
+
+  public static void clearElementValue(WebElement element) {
+    scrollToElement(element);
+    WaitHelper.waitForElementToBeDisplayed(element);
+    logger.info("Clear WebElement: " + element);
+    element.click();
+    element.clear();
+    while (!element.getAttribute("value").equals("")) {
+      element.sendKeys(Keys.BACK_SPACE);
+    }
+  }
+
+  /**
+   * Replacing the value of the text box when clear is not working
+   * https://github.com/SeleniumHQ/selenium/issues/6741
+   */
+  public static void replaceElementValue(WebElement element, String value) {
+    replaceElementValue(element, value, 1024);
+  }
+
+  public static void replaceElementValue(WebElement element, String value, int limit) {
+    WaitHelper.waitForElementToBeDisplayed(element);
+    boolean flag = false;
+    for (int i = 0; i < limit; i++) {
+      if (element.getAttribute("value").equals(StringUtils.EMPTY)) {
+        flag = true;
+        break;
+      }
+      element.sendKeys(Keys.BACK_SPACE);
+    }
+    if (!flag) {
+      Assert.fail("Element's current value is not cleared - " + element + " - "
+        + "Current value :" + element.getAttribute("value"));
+    }
+    element.sendKeys(value);
+  }
+
+  public static void dragAndDrop(WebElement source, WebElement target) {
+    WaitHelper.waitForElementToBeDisplayed(source);
+    WaitHelper.waitForElementToBeDisplayed(target);
+    logger.info("Drag and drop between source: " + source + " and target: " + target);
+    actions.dragAndDrop(source, target).build().perform();
+  }
+
+  /**
+   * Performs click-and-hold at the location of the element, moves by a given offset, then releases the mouse.
+   */
+  public static void dragAndDropByOffset(WebElement element, int xOffset, int yOffset) {
```

Comment:
Why not waiting for element to be displayed in this method? Although seems unlikely to drag and drop an element which is not present but would be good to re-assure.

Reply:
Updated